src/data/vapor-thermal-conductivity-constants.ts

```diff
@@ -0,0 +1,766 @@
+  },
+
+  secButylMercaptan: {
+    name: "sec-Butyl mercaptan",
+    formula: "C4H10S",
+    CAS: "513-53-1",
+    molecularWeight: 90.1872,
+    C1: 0.9719,
+    C2: -0.111,
+    C3: 1167.2,
+    C4: 3163200,
+    minimumTemperature: 358.13,
+    maximumTemperature: 1000,
+    thermalConductivityAtMinimumTemperature: 0.01749,
+    thermalConductivityAtMaximumTemperature: 0.0847
+  },
+
+  "1Butyne": {
+    name: "1-Butyne",
+    formula: "C4H6",
+    CAS: "107-00-6",
+    molecularWeight: 54.09044,
+    C1: 3.7269e-5,
+    C2: 1.1427,
+    C3: -43.844,
+    C4: 79421,
+    minimumTemperature: 281.22,
+    maximumTemperature: 1000,
+    thermalConductivityAtMinimumTemperature: 0.01268,
+    thermalConductivityAtMaximumTemperature: 0.09644
+  },
+
+  Butyraldehyde: {
+    name: "Butyraldehyde",
+    formula: "C4H8O",
+    CAS: "123-72-8",
+    molecularWeight: 72.10572,
+    C1: 9.9652e-7,
+    C2: 1.6558,
+    minimumTemperature: 347.94,
+    maximumTemperature: 1000,
+    thermalConductivityAtMinimumTemperature: 0.0161,
+    thermalConductivityAtMaximumTemperature: 0.09245
+  },
+
+  ButyricAcid: {
+    name: "Butyric acid",
+    formula: "C4H8O2",
+    CAS: "107-92-6",
+    molecularWeight: 88.1051,
+    C1: 0.7873,
+    C2: -0.0036161,
+    C3: 5.6641e-6,
+    C4: -2.8451e-9,
+    minimumTemperature: 436.42,
+    maximumTemperature: 706.95,
+    thermalConductivityAtMinimumTemperature: 0.05147,
+    thermalConductivityAtMaximumTemperature: 0.05647,
+  },
+
+  ButyricAcidExtended: {
+    name: "Butyric acid",
+    formula: "C4H8O2",
+    CAS: "107-92-6",
+    molecularWeight: 88.1051,
+    C1: 9.2069e-8,
+    C2: 2.0312,
+    minimumTemperature: 706.95,
+    maximumTemperature: 1000,
+    thermalConductivityAtMinimumTemperature: 0.05647,
+    thermalConductivityAtMaximumTemperature: 0.11421,
+  },
+
+  Butyronitrile: {
+    name: "Butyronitrile",
+    formula: "C4H7N",
+    CAS: "109-74-0",
+    molecularWeight: 69.1051,
+    C1: 1.3751e-6,
+    C2: 1.5786,
+    minimumTemperature: 390.74,
+    maximumTemperature: 1000,
+    thermalConductivityAtMinimumTemperature: 0.01698,
+    thermalConductivityAtMaximumTemperature: 0.07484
+  },
+
+  CarbonDioxide: {
+    name: "Carbon dioxide",
+    formula: "CO2",
+    CAS: "124-38-9",
+    molecularWeight: 44.0095,
+    C1: 3.69,
+    C2: -0.3838,
+    C3: 964,
+    C4: 1860000,
+    minimumTemperature: 194.67,
+    maximumTemperature: 1500,
+    thermalConductivityAtMinimumTemperature: 0.00887,
+    thermalConductivityAtMaximumTemperature: 0.09025
+  }
+};
+
+export function calculateDIPPRVaporThermalConductivity(compound: string, temperature: number): number {
+  const vaporThermalConductivityProps: DIPPRVaporThermalConductivityProps | undefined =
+    DIPPRVaporThermalConductivityDictionary[compound];
+  let thermalConductivity = NaN;
+  let compoundExceptionList = ["AceticAcid","AceticAcidExtended1", "ButyricAcid", "FormicAcid", "HeptanoicAcid", "OctanoicAcid", "PentanoicAcid", "PropionicAcid"];
```

Comment:
- Why is `let` used?
- Please run linter here. Seems like there's some formatting issue.